### PR TITLE
adapt to ClientCommunicator api changes

### PR DESCRIPTION
**Pull request by chrisdennis**

### CODE REVIEW

File: proxy/src/test/java/org/terracotta/voltron/proxy/EndToEndTest.java [hunk 1]
```diff
@@ -155,14 +157,15 @@ public void onMessage(final Integer message) {
...
+      public void sendNoResponse(final ClientDescriptor clientDescriptor, final EntityResponse message) {
         throw new UnsupportedOperationException("Implement me!");
       }
 
-      public Future<Void> send(final ClientDescriptor clientDescriptor, final byte[] bytes) {
+      public Future<Void> send(final ClientDescriptor clientDescriptor, final EntityResponse message) {
         final FutureTask<Void> voidFutureTask = new FutureTask<Void>(new Callable<Void>() {
           public Void call() throws Exception {
-            listener.onMessage((Integer)codec.decode(Arrays.copyOfRange(bytes, 1, bytes.length), Integer.class));
+            ProxyEntityResponse pem = (ProxyEntityResponse) message;
```
Comment (anthonydahanne) [inline proxy/src/test/java/org/terracotta/voltron/proxy/EndToEndTest.java]: pem ? meh, nevermind

File: proxy/src/test/java/org/terracotta/voltron/proxy/client/VoltronProxyInvocationHandlerTest.java [hunk 2]
```diff
@@ -26,14 +26,22 @@
...
 import java.util.concurrent.Future;
 
 import static org.junit.Assert.*;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
+import org.terracotta.voltron.proxy.CommonProxyFactory;
+import static org.terracotta.voltron.proxy.CommonProxyFactory.createMethodMappings;
+import static org.terracotta.voltron.proxy.CommonProxyFactory.invert;
+import org.terracotta.voltron.proxy.ProxyMessageCodec;
+import org.terracotta.voltron.proxy.server.messages.ProxyEntityResponse;
```
Comment (anthonydahanne) [inline proxy/src/test/java/org/terracotta/voltron/proxy/client/VoltronProxyInvocationHandlerTest.java]: some imports are not used 
import static java.util.Collections.emptyMap;
import java.util.HashMap;
import java.util.concurrent.Future;
import org.terracotta.voltron.proxy.CommonProxyFactory;

**Comment by anthonydahanne** [comment]
all set !
waiting for imports update before merging !

**Comment by chrisdennis** [comment]
done

**Pull request merged**